siliconcompiler/libs/nangate45.py

```diff
@@ -0,0 +1,104 @@
+
+import os
+import sys
+import re
+import siliconcompiler
+
+def make_docs():
+    '''
+    Library description
+
+    Documentation: https://
+    Sources: https://
+    Installation: https://
+    '''
+
+    chip = siliconcompiler.Chip()
+    setup_lib(chip)
+    return chip
+
+def setup(chip):
+
+    libname = 'nangate45'
+    foundry = 'virtual'
+    process = 'freepdk45'
+    libtype = '10t'
+    rev = 'r1p0'
+    corner = 'typical'
+    objectives = ['setup']
+
+    libdir = os.path.join('..',
+                          'third_party',
+                          'pdks',
+                          foundry,
+                          process,
+                          'libs',
+                          libname,
+                          rev)
+
+    # standard cell typ
+    chip.set('library',libname,'type','stdcell')
+
+    # rev
+    chip.set('library',libname, 'package', 'version',rev)
+
+    # timing
+    chip.add('library',libname, 'nldm', corner, 'lib',
```

Comment:
Ditto the `asap7.py` commetn; `.add` or `.set`? (And ditto throughout this file)